### ISSUE TITLE
Formatting rule for disable directive spacing

### ISSUE THREAD

**Issue by TSMMark** [eslint/eslint#14878]
**Please describe what the rule should do:**

warn about (and ideally fix) spacing in eslint-disable and eslint-enable comments.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Bad: 

```js
/* eslint-disable   some-rule */
/* eslint-enable   some-rule */
/* eslint-disable some-rule,   other-rule    */
//    eslint-disable-next-line   some-rule,   other-rule
// eslint-disable-line   some-rule   ,   other-rule
```

Good: 

```js
/* eslint-disable some-rule */
/* eslint-enable some-rule */
/* eslint-disable some-rule, other-rule */
// eslint-disable-next-line some-rule, other-rule */
// eslint-disable-line some-rule, other-rule */
```

**Why should this rule be included in ESLint (instead of a plugin)?**

#14617 introduces fixable disable directives, which have the potential to be formatted with awkward spacing, so [@mdjermanovic recommended opening this new issue](https://github.com/eslint/eslint/pull/14617#issuecomment-892192785)

**Are you willing to submit a pull request to implement this rule?**

Potentially

**Comment by nzakas** [comment]
I’m not in favor of this. With nearly 300 rules in the core already, I don’t think this rises to the level needed for inclusion. 

You can always create a custom rule to do this if it’s important to you.

**Comment by bmish** [comment]
[eslint-plugin-eslint-comments](https://www.npmjs.com/package/eslint-plugin-eslint-comments) could be a good place for such a rule.

**Comment by mdjermanovic** [comment]
@TSMMark thanks for opening the issue! I don't think we should add a core rule for this (plugin is a good idea though). My comment https://github.com/eslint/eslint/pull/14617#issuecomment-892192785 was about updating the autofix we implemented in https://github.com/eslint/eslint/pull/14617 to better handle the leading space if it doesn't end up being too complex.

**Comment by nzakas** [comment]
I’m 👍 to trying to produce a better autofix

**Comment by mdjermanovic** [comment]
> #14617 introduces fixable disable directives, which have the potential to be formatted with awkward spacing

I'm working on https://github.com/eslint/eslint/pull/14617#issuecomment-892105159

**Comment by mdjermanovic** [comment]
I'm closing this issue since it isn't likely that the proposed rule will be accepted.

Fix for https://github.com/eslint/eslint/pull/14617#issuecomment-892105159 is included in the recent [ESLint v8.0.0-rc.0 release](https://eslint.org/blog/2021/09/eslint-v8.0.0-rc.0-released).

@TSMMark thanks for testing new features on your codebase! Please let us know if you encounter any issues with the new version of this autofix.
